src/MongoDB.Driver.Core/Core/Connections/IConnection.cs

```diff
@@ -81,6 +78,12 @@ public interface IConnection : IDisposable
         /// </value>
         ConnectionSettings Settings { get; }
 
+        /// <summary>
+        /// Add additional read timeout value.
+        /// </summary>
+        /// <param name="timeout">The timeout.</param>
+        void AddAdditionalReadTimeout(TimeSpan timeout);
```

Comment:
Remove this method.

The timeout should be adjusted directly on the connection.

Reply:
I mean replace this with `SetReadTimeout`.

Not add some value to an unknown current timeout.

Reply:
Will change it

Reply:
As long as the behavior is preserved. We do need to set the read timeout to the connection's original read timeout (which in turn is equal to the connect timeout from the MongoClient) plus the heartbeat frequency